user-interface/src/lib/components/combobox/ComboBox.test.tsx

```diff
@@ -0,0 +1,692 @@
+    });
+  });
+
+  test('If the dropdown list is open, and the input field is in focus, then pressing the Tab key should highlight the first item in the dropdown list. Pressing the tab key a second time should close the dropdown and focus on the next element on the screen.', async () => {
+    renderWithProps();
+
+    const comboboxInputField = focusComboInputField(comboboxId);
+    expect(isDropdownClosed()).toBeFalsy();
+
+    fireEvent.keyDown(comboboxInputField!, { key: 'Tab' });
+
+    const listItem = document.querySelector('li button');
+    await waitFor(() => {
+      expect(listItem).toHaveFocus();
+    });
+    expect(isDropdownClosed()).toBeFalsy();
+
+    await userEvent.tab();
+
+    await waitFor(() => {
+      expect(isDropdownClosed()).toBeTruthy();
+      const input1 = document.querySelector('.input1');
+      expect(input1).toHaveFocus();
+    });
+  });
+
+  test('If the dropdown is already closed, then pressing the tab key should focus on the next element on the screen and should not open the dropdown list.', async () => {
+    renderWithProps();
+
+    const otherButton = document.querySelector('.button1');
+    const otherInput = document.querySelector('.input1');
+    const comboboxInputField = document.querySelector(`#${comboboxId}-combo-box-input`);
+    (otherButton as HTMLButtonElement)!.focus();
+    expect(otherButton!).toHaveFocus();
+    await userEvent.tab();
+    await waitFor(() => {
+      expect(comboboxInputField!).toHaveFocus();
+    });
+    expect(isDropdownClosed()).toBeTruthy();
+
+    await userEvent.tab();
+    expect(otherInput!).toHaveFocus();
+  });
+
+  test('If the dropdown is open, and gibberish is typed into input field, the filtered results should have length of 0, and pressing tab should tab out of the combobox to the next input field.', async () => {
+    const ref = React.createRef<ComboBoxRef>();
+    renderWithProps({}, ref);
+
+    const otherInput = document.querySelector('.input1');
+    const comboboxInputField = focusComboInputField(comboboxId);
+
+    fireEvent.change(comboboxInputField!, { target: { value: 'this is gibberish' } });
+    (comboboxInputField! as HTMLInputElement).focus();
+    const result = ref.current?.getValue();
+    expect(result).toEqual([]);
+
+    await userEvent.tab();
+    await waitFor(() => {
+      expect(otherInput!).toHaveFocus();
+    });
+    expect(isDropdownClosed()).toBeTruthy();
+  });
+
+  test('If the dropdown list is closed, then typing characters on the keyboard should open the dropdown list', async () => {
+    renderWithProps();
+
+    const inputField = focusComboInputField(comboboxId);
+    if (inputField) {
+      fireEvent.keyDown(inputField, { key: 'Escape' });
+      expect(isDropdownClosed()).toBeTruthy();
+
+      fireEvent.change(inputField, { target: { value: 'test input' } });
+      expect(isDropdownClosed()).toBeFalsy();
+    } else {
+      throw new Error('input field not found');
```

Comment:
Why not add `expect(inputField).toBeTruthy();` after line 308? Or maybe throw it all in a `waitFor`?